From b8ae8f713a15b00ec447a23294948fc463220130 Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Mon, 6 May 2024 21:58:09 +0100
Subject: [PATCH] screencast: Correct expected bus name for streams
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Previously, this only worked because GDBusConnection was not filtering
signals by their sender correctly (GNOME/glib#3268).

Thanks: Alicia Boya García
Signed-off-by: Simon McVittie <smcv@debian.org>
Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/3303>
---
 js/dbusServices/screencast/screencastService.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/dbusServices/screencast/screencastService.js b/js/dbusServices/screencast/screencastService.js
index 708a15c84..0321b4d69 100644
--- a/js/dbusServices/screencast/screencastService.js
+++ b/js/dbusServices/screencast/screencastService.js
@@ -161,7 +161,7 @@ var Recorder = class {
             });
 
         this._streamProxy = new ScreenCastStreamProxy(Gio.DBus.session,
-            'org.gnome.ScreenCast.Stream',
+            'org.gnome.Mutter.ScreenCast',
             streamPath);
 
         this._streamProxy.connectSignal('PipeWireStreamAdded',
-- 
2.45.0

